lib/generators/hyku_knapsack/work/work_generator.rb

```diff
@@ -0,0 +1,142 @@
+# frozen_string_literal: true
+
+# This is a copy of the Hyrax work generate adapted for use in the Knapsack.
+# example:
+#   `rails generate hyku_knapsack:work WorkType`
+#
+# This will ensure the generated files get created in the knapsack and not the submodule
+
+require 'rails/generators'
+require 'rails/generators/model_helpers'
+
+class HykuKnapsack::WorkGenerator < Rails::Generators::NamedBase
+  # ActiveSupport can interpret models as plural which causes
+  # counter-intuitive route paths. Pull in ModelHelpers from
+  # Rails which warns users about pluralization when generating
+  # new models or scaffolds.
+  include Rails::Generators::ModelHelpers
+
+  source_root File.expand_path('../templates', __FILE__)
+
+  argument :attributes, type: :array, default: [], banner: 'field:type field:type'
+
+  TEMPLATE_PATH = Hyrax::Engine.root.join('lib', 'generators', 'hyrax', 'work', 'templates')
```

Comment:
I'm confused, how is this used compared to source_root?

Reply:
Since we're moving this file from the Hyrax gem's `lib/generators/hyrax/` folder, it doesn't know where all of its template files are because we're not in that dir structure.  This line is to tell the generator where it's at.  Normally this file isn't in Hyku at all, it stays in Hyrax.